### PR TITLE
Implement SSL and TLS Config

### PR DESCRIPTION
**Pull request by minitriga**
Related Issue: #293 

## New Behavior

Set SSL and TLS configuration. 

## Contrast to Current Behavior

SSL and TLS config does not exist. 

## Discussion: Benefits and Drawbacks

No Drawbacks. 

## Changes to the Wiki

None

## Proposed Release Note Entry

Implement SSL and TLS Environment Variables.

## Double Check

* [x] I have read the comments and followed the PR template.
* [x] I have explained my PR according to the information in the comments.
* [x] My PR targets the `develop` branch.

### CODE REVIEW

File: configuration/configuration.py [hunk 1]
```diff
@@ -119,6 +119,10 @@ def read_secret(secret_name):
     'PASSWORD': os.environ.get('EMAIL_PASSWORD', read_secret('email_password')),
     'TIMEOUT': int(os.environ.get('EMAIL_TIMEOUT', 10)),  # seconds
     'FROM_EMAIL': os.environ.get('EMAIL_FROM', ''),
+    'USE_SSL': os.environ.get('EMAIL_USE_SSL', 'False'),
+    'USE_TLS': os.environ.get('EMAIL_USE_TLS', 'False'),
```
Comment (tobiasge) [inline configuration/configuration.py]: You're settings `USE_SSL` and `USE_TLS` to a string, but it should be a boolean. Have a look at how it is done with `METRICS_ENABLED`.

File: env/netbox.env [hunk 2]
```diff
@@ -9,6 +9,10 @@ EMAIL_USERNAME=netbox
 EMAIL_PASSWORD=
 EMAIL_TIMEOUT=5
 EMAIL_FROM=netbox@bar.com
+EMAIL_USE_SSL=false
+EMAIL_USE_TLS=false
```
Comment (cimnine) [inline env/netbox.env:14]:
```suggestion
# EMAIL_USE_SSL and EMAIL_USE_TLS are mutually exclusive, i.e. they can't both be `true`!
EMAIL_USE_SSL=false
EMAIL_USE_TLS=false
```

**Review by cimnine — CHANGES_REQUESTED** [review]
Thanks for the PR. Just a tiny suggestion to reduce the possibility of support issues being created.

Also, @tobiasge's comment.

File: env/netbox.env (hunk 2 above)
Comment (minitriga) [reply env/netbox.env:14]: Implemented this change.

File: configuration/configuration.py (hunk 1 above)
Comment (minitriga) [reply configuration/configuration.py]: Implemented. Thanks for the advice.

**Comment by minitriga** [comment]
@cimnine Updated Thanks for the assistance.

**Review by cimnine — APPROVED** [review]

**Comment by yngmike** [comment]
Was this ever fixed?, I can't seem to find anywhere to set the SSL certificate.
I've seen comments to use a reverse proxy etc... I really believe SSL support should be  bundled into the docker container.
With the current level of security craziness out there running anything without SSL is just asking to get compromised.
Not to mention my security guy is going to find it in a scan and block it in the IPS.

**Comment by kkthxbye-code** [comment]
@yngmike - this old PR is regarding connecting to mail servers with TLS, not the HTTP server.

>With the current level of security craziness out there running anything without SSL is just asking to get compromised.

It's fairly uncommon in production container setups to expose the container directly instead of trough an ingress controller like nginx ingress for kubernetes or a reverse proxy like traefik/nginx/caddy etc. if using docker. The ingress controller/reverse proxy would handle TLS as you mention.

>I really believe SSL support should be bundled into the docker container.

You should raise this in a discussion instead and actually give reasons why you believe it should be included in the container.

**Pull request merged**